plugins/PostProcessingPlugin/scripts/InsertAtLayerChange.py

```diff
@@ -1,65 +1,218 @@
                 {
-                    "label": "When to insert",
-                    "description": "Whether to insert code before or after layer change.",
+                    "label": "Enable this script",
+                    "description": "You must enable the script for it to run.",
+                    "type": "bool",
+                    "default_value": true,
+                    "enabled": true
+                },
+                "insert_frequency":
+                {
+                    "label": "How often to insert",
+                    "description": "Every so many layers starting with the Start Layer OR as single insertion at a specific layer.  If the print sequence is 'one_at_a_time' then the insertions will be made for every model.  Insertions are made at the beginning of a layer.",
                     "type": "enum",
-                    "options": {"before": "Before", "after": "After"},
-                    "default_value": "before"
+                    "options": {
+                        "once_only": "One insertion only",
+                        "every_layer": "Every Layer",
+                        "every_2nd": "Every 2nd",
+                        "every_3rd": "Every 3rd",
+                        "every_5th": "Every 5th",
+                        "every_10th": "Every 10th",
+                        "every_25th": "Every 25th",
+                        "every_50th": "Every 50th",
+                        "every_100th": "Every 100th"},
+                    "default_value": "every_layer",
+                    "enabled": "enable_insatlaychange"
                 },
-                "gcode_to_add":
+                "start_layer":
                 {
-                    "label": "G-code to insert",
-                    "description": "G-code to add before or after layer change.",
-                    "type": "str",
-                    "default_value": ""
+                    "label": "Starting Layer",
+                    "description": "Layer to start the insertion at.  If the Print_Sequence is 'All at Once' then use the layer numbers from the Cura Preview.  Enter '1' to start at gcode LAYER:0. In 'One at a Time' mode use the layer numbers from the first model that prints AND all models will receive the same insertions.  NOTE: There is never an insertion for raft layers.",
+                    "type": "int",
+                    "default_value": 1,
+                    "minimum_value": 1,
+                    "enabled": "insert_frequency != 'once_only' and enable_insatlaychange"
                 },
-                "skip_layers":
+                "end_layer":
                 {
-                    "label": "Skip layers",
-                    "description": "Number of layers to skip between insertions (0 for every layer).",
+                    "label": "Ending Layer",
+                    "description": "Layer to end the insertion at. Enter '-1' to indicate the entire file.  Use layer numbers from the Cura Preview.",
```

Comment:
Same here, can we indicate if the last insertion will occur before or after this layer ?